tests/network/libs/connectivity.py

```
@@ -66,3 +72,21 @@ def poll_tcp_connectivity(
+    Args:
+        ping_output: The full output of a ping command.
+
+    Returns:
+        The packet-loss percentage (0-100) reported in the statistics line.
+
+    Raises:
+        PacketLossSummaryNotFoundError: If the ping output has no packet-loss summary line.
+    """
+    match = re.search(pattern=r"(\d+(?:\.\d+)?)% packet loss", string=ping_output)
+    if not match:
+        raise PacketLossSummaryNotFoundError(f"No packet-loss summary found in ping output: {ping_output}")
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)